tests/test_shell.py

```diff
@@ -113,9 +130,13 @@ def test_statepoint(self):
         self.call('python -m signac init my_project'.split())
         self.call(['python', '-m', 'signac', 'job', '--create', '{"a": 0}'])
         project = signac.Project()
-        for job in project:
-            sp = self.call('python -m signac statepoint {}'.format(job).split())
-            assert project.open_job(json.loads(sp)) == job
+        job = project.open_job({'a': 0})
+        sp = self.call('python -m signac statepoint {}'.format(job).split())
+        assert project.open_job(json.loads(sp)) == job
+        sp = self.call('python -m signac statepoint'.split())
+        assert project.open_job(json.loads(sp)) == job
+        sp = self.call('python -m signac statepoint --pretty'.split())
+        assert "{'a': 0}" in sp
```

Comment:
I found some inconsistent behavior. In general, I believe the signac CLI should always have JSON-compatible output where possible. In a test project, I ran the following:
```bash
$ signac statepoint
{"a": {"b": 0, "c": 0}}
$ signac statepoint --pretty
{'a': {'b': 0, 'c': 0}}
```
The `--pretty` output is not JSON-compatible because JSON requires double quotes, not single quotes. @csadorf Is this something we've thought about before? I don't know how `--pretty` is implemented.

Reply:
We use `pprint` to implement pretty. We should probably have used some JSON prettifying instead. Feel free to classify as bug and fix without deprecation cycle.

Reply:
Switching from pprint to something like `json.dumps` would also give us something closer to the output that @phmalek requested recently on Gitter

Reply:
While `json.dumps` offers an `indent`, there is no way to have the equivalent of pprint's `depth` flag with `json.dumps`:

```python
>>> from pprint import pprint
>>> pprint({'a': {'very': {'deep': 'dictionary'}}})
{'a': {'very': {'deep': 'dictionary'}}}
>>> pprint({'a': {'very': {'deep': 'dictionary'}}}, depth=2)
{'a': {'very': {...}}}
```
@csadorf @vyasr Is this "depth" a hard requirement for pretty output?

Reply:
I did find that kind of important in my interactive work, yes.

Reply:
In that case, I recommend to make no changes to existing behavior. If "pretty" output was just about indenting it, then that can be solved with `json.dumps`. Since there's not an easy way to solve this, let's leave the current behavior as-is. If you agree, we can resolve the conversations about pretty output.